### PR TITLE
Allow VectorElement to take a variant argument.

### PR DESCRIPTION
**Pull request by pefarrell**

### CODE REVIEW

**Comment by wence-** [comment]
Can you do this for `TensorElement` too, then I'll merge.

File: ufl/finiteelement/mixedelement.py [hunk 1]
```diff
@@ -263,6 +263,8 @@ def __init__(self, family, cell=None, degree=None, dim=None,
                viewed as k-form), ignored if family is a FiniteElement
             quad_scheme
                The quadrature scheme (optional), ignored if family is a FiniteElement
+            variant
```
Comment (garth-wells) [inline ufl/finiteelement/mixedelement.py:266]: I think this needs a bit more explanation. I don't understand the docstring.

Comment (miklos1) [reply ufl/finiteelement/mixedelement.py:266]: This variant thing was first introduced to distinguish between GL, GLL, and equispaced Langrange elements without a profusion of names: Q, dQ, RTCE, RTCF, NCE, NCF type elements could be built from equispaced Lagrange as well as from spectral ingredients.

Later uses of this include RT and N1curl elements being available with different nodal basis, etc.

Comment (miklos1) [reply ufl/finiteelement/mixedelement.py:266]: c.f. FEniCS/fiat#52

Comment (miklos1) [reply ufl/finiteelement/mixedelement.py:266]: For further reference, see [this UFL issue](https://bitbucket.org/fenics-project/ufl/issues/92) and [this UFL pull request](https://bitbucket.org/fenics-project/ufl/pull-requests/69).

**Pull request merged**